File: python/026-darts/darts.py
import numbers

def radius(dart_x, dart_y):
    if not all(isinstance(coord, numbers.Number) for coord in (dart_x, dart_y)):
        raise TypeError("Not 2 numbers")
    return dart_x * dart_x + dart_y * dart_y

def score(dart_x, dart_y):
    rad = radius(dart_x, dart_y)
    if rad > 100:
        return 0
    if rad > 25:
        return 1
    if rad > 1:
        return 5
    return 10

File: typescript/011-darts/darts.ts
function radiusSquared(x: number, y: number): number {
  return x ** 2 + y ** 2;
}

export function score(x: number, y: number): number {
  const radSq = radiusSquared(x, y);
  if (radSq > 10 ** 2) {
    return 0
  } else if (radSq > 5 ** 2) {
    return 1
  } else if (radSq > 1 ** 2) {
    return 5
  } else {
    return 10
  }
}
